tests/Microsoft.Identity.Test.Unit.net45/OAuthClientTests.cs

```diff
@@ -164,6 +167,54 @@ public void OAuthClient_FailsWithServiceExceptionWhenResponseIsHttpNotFound()
                 });
         }
 
+        [TestMethod]
+        public void PKeyAuthSuccsesResponseTest()
```

Comment:
There 3 types of tests that you need to write / modify: 

1. Integration tests - these are great to ensure that EVO keeps it's part of the contract. I don't think you need to write new tests, you just need to make sure the HTTP traffic is happening as you expect - with PKAuth challenges and headers. You can use this class to sniff the HTTP traffic in integration tests https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/commit/1c03719723c125d94399c6496dd6db9d988a93a4

2. End to End unit tests like this one - these are great if you are able to mock IDeviceManager (see how we've mocked IBroker for an example). This allows you to test the OAuth2Client by manipualting the traffic. Potentially not needed if you move more logic to IDeviceManager

3. Conventional Unit tests which look at objects in isolation. In this case, the various implementations of IDeviceManager, especially the .net classic one needs to be tested. Consider starting with a well know public / private key pair and asserting the header values.